### DISCUSSION TITLE
Time Series models with time-varying past covariates

### DISCUSSION THREAD

**Discussion by bustosxaus** [autogluon/autogluon#3134] [Q&A] [answered]
Hi everyone,

I am currently looking for forecasting models that can handle time-varying past covariates. From the documentation, it appears that the following models in AutoGluon Time Series are capable of supporting past continuous covariates: AutoGluonTabularModel, TemporalFusionTransformerModel and MQCNNMXNetModel. I have a few questions regarding these models:

- Is AutoGluonTabularModel available with medium_quality and high_quality presets, but not with best_quality?

- How can I try using MQCNNMXNetModel? I attempted to use the following code, but the model was not supported:

```python
predictor.fit(
    train_data,
    hyperparameters={
        "MQCNNMXNetModel": {},
    }
)
```

- Are there plans to introduce more models that can support past covariates in the future?

Thank you in advance for any advice or help you can provide me.

**Comment by shchur** [comment]
Hi, 
1. This should have been fixed on the master branch (see #2997).
2. Most likely, you get this error because MXNet is not installed, which is necessary to run this model. See https://auto.gluon.ai/new/tutorials/timeseries/forecasting-model-zoo.html#mxnet-models for the instructions.
3. Yes, we plan to continue adding models to AG-TS, but I cannot make any promises about the specific time line for this.

**Comment by bustosxaus** [comment]
Thank you for the quick reply, Oleksandr, regarding 2 I do have MXNet installed, I am able to run `TransformerMXNet` when using `best_quality`, but I only have the CPU version installed, could this be the problem? Thank you!

**Reply by shchur** [reply, marked as answer]
Oh, I see now, you need to change the name to `"MQCNNMXNet"`

**Comment by bustosxaus** [comment]
Thank you! It worked!
